### PR TITLE
enhancement(deps): Split fake data into separate module

### PR DESCRIPTION
**Pull request by lucperkins**
Just a little bit of codebase housekeeping. This splits off the fake log generator helper library into a separate module. This makes it usable in places outside of `sources`, for example in benchmarking and testing. Eventually I'd like to use it to provide parameterized fake metrics as well.

### CODE REVIEW

File: lib/fakedata/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+name = "fakedata"
+version = "0.1.0"
+authors = ["Vector Contributors <vector@timber.io>"]
+edition = "2018"
+publish = false
+license = "MIT"
```
Comment (ktff) [inline lib/fakedata/Cargo.toml:7]: License should be "MPL-2.0"

File: lib/fakedata/src/logs.rs [hunk 2]
```diff
@@ -0,0 +1,220 @@
+use chrono::{prelude::Local, SecondsFormat};
```
Comment (ktff) [inline lib/fakedata/src/logs.rs:1]: Is the original file deleted?

Comment (lucperkins) [reply lib/fakedata/src/logs.rs:1]: This is required for the `timestamp_*` functions

Comment (lucperkins) [reply lib/fakedata/src/logs.rs]: Nope! My bad. It is now.

**Review by blt — APPROVED** [review]

**Comment by lucperkins** [comment]
The test failure here appears to be spurious (`sources::file::file_rotate`), so I'm going to merge

**Review by ktff — COMMENTED** [review]
LGTM

**Pull request merged**